logr.go

```diff
@@ -188,6 +191,13 @@ type Logger struct {
 	level         int
 	sink          LogSink
 	withCallDepth CallDepthLogSink
+
+	// withHelper is set in Init if the log sink supports the interface.
```

Comment:
I'm going to be super nit-picky to avoid followups :)

```
	// withHelper is set during init if the log sink supports the HelperLogSink
	// interface.
	//
	// We only need the function returned by withHelper.GetHelper(), but
	// cannot store it here because that makes Logger non-comparable,
	// which is something we want to guarantee.
```